core/src/bmi.jl

```diff
@@ -105,7 +105,14 @@ function BMI.initialize(T::Type{Model}, config::Config)::Model
     end
     @debug "Setup ODEProblem."
 
-    callback, saved_flow = create_callbacks(parameters)
+    # expand scalar saveat for SavingCallback
+    saveat = if config.solver.saveat isa Number
+        range(timespan...; step = config.solver.saveat)
+    else
+        config.solver.saveat
```

Comment:
Should there be a check that this is indeed a list of floats if it is not a number?